crates/autopilot/src/run.rs

```diff
@@ -510,7 +511,13 @@ pub async fn run(args: Arguments) {
         .await
         .expect("Should be able to initialize event updater. Database read issues?");
 
-        maintenance.with_ethflow(onchain_order_indexer, refund_event_handler);
+        maintenance.with_ethflow(onchain_order_indexer);
+        // refunds are not critical for correctness and can therefore be indexed
+        // sporadically in a background task
+        let service_maintainer = ServiceMaintenance::new(vec![Arc::new(refund_event_handler)]);
```

Comment:
We already have logic inside `maintenance.rs` that runs on every new block even in the synchronized mode. Maybe it's better to keep this logic there?